services/merak-topo/handler/handler.go

```diff
@@ -15,70 +15,89 @@ package handler
 
 import (
 	"context"
-	"log"
-
+	"errors"
+	"strings"
 	"time"
 
 	"github.com/futurewei-cloud/merak/services/merak-topo/database"
 
-	"fmt"
-
-	"strings"
-
 	pb_common "github.com/futurewei-cloud/merak/api/proto/v1/common"
 	pb "github.com/futurewei-cloud/merak/api/proto/v1/topology"
+	"github.com/futurewei-cloud/merak/services/merak-topo/utils"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
 //function CREATE
 /* save the part of gw creation and mac learning for future requirment, comment the related code now*/
-func Create(k8client *kubernetes.Clientset, topo_id string, aca_num uint32, rack_num uint32, aca_per_rack uint32, cgw_num uint32, data_plane_cidr string, ports_per_vswitch uint32, images []*pb.InternalTopologyImage, aca_parameters string, returnMessage *pb.ReturnTopologyMessage) error {
+func Create(k8client *kubernetes.Clientset, topo_id string, aca_num uint32, rack_num uint32, aca_per_rack uint32, cgw_num uint32, data_plane_cidr string, ports_per_vswitch uint32, images []*pb.InternalTopologyImage, aca_parameters string, returnMessage *pb.ReturnTopologyMessage, topoPrefix string, namespace string) error {
 
 	start_time := time.Now()
-	log.Println("CREATE: ====== Start: Parse gRPC message ======= ")
-	log.Printf("Vhost number is: %v\n", aca_num)
-	log.Printf("Rack number is: %v\n", rack_num)
-	log.Printf("Vhosts per rack is: %v\n", aca_per_rack)
 
-	log.Printf("Ports per vswitch is: %v\n", ports_per_vswitch)
+	var aca_image string
+	var ovs_image string
+	var err_return error
+
+	err_flag := 0
+	errs := errors.New("request DEPLOY- fails to create topology")
+
+	for _, img := range images {
+		if strings.Contains(img.Name, "ACA") {
+			aca_image = img.Registry
+		} else if strings.Contains(img.Name, "OVS") {
+			ovs_image = img.Registry
+		}
+	}
 
-	log.Println("CREATE:====== Start: Generate multi-layer topology structure ======= ")
+	utils.Logger.Debug("request DEPLOY details", "Vhost number", aca_num, "Rack number", rack_num, "Vhosts per rack", aca_per_rack, "Ports per vswitch", ports_per_vswitch)
 
-	err_create, topo := Create_multiple_layers_vswitches(int(aca_num), int(rack_num), int(aca_per_rack), int(ports_per_vswitch), data_plane_cidr)
+	topo, err_create := Create_multiple_layers_vswitches(int(aca_num), int(rack_num), int(aca_per_rack), int(ports_per_vswitch), data_plane_cidr)
 	if err_create != nil {
-		return fmt.Errorf("create multiple layers vswitches error %s", err_create)
+		utils.Logger.Error("request DEPLOY", "multiple layers vswitches", err_create.Error())
+		returnMessage.ReturnMessage = "Can not create multiple layers vswitches"
+		returnMessage.ReturnCode = pb_common.ReturnCode_FAILED
+		err_flag = 1
+
```

Comment:
@zzengcs Is that mean even we have error from `Create_multiple_layers_vswitches`, the logic here still go on for the rest of process? Is that safe?